src/overlays/actors/ovl_Bg_Last_Bwall/z_bg_last_bwall.c

```diff
@@ -28,57 +37,210 @@ const ActorInit Bg_Last_Bwall_InitVars = {
+static s16 D_80C18A78[] = { 0, 1, 2, 0, 2, 3 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/func_80C187F8.s")
+static InitColliderStruct D_80C18A84 = { D_80C18A60, D_80C18A78, 2 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/func_80C1886C.s")
+static Vec3s D_80C18A90[] = {
+    { -0x50, 0x190, -0x50 },
+    { -0x50, 0x190, 0x50 },
+    { 0x50, 0x190, 0x50 },
+    { 0x50, 0x190, -0x50 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/func_80C18884.s")
+static s16 D_80C18AA8[] = { 0, 3, 2, 0, 2, 1 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/func_80C188C4.s")
+static InitColliderStruct D_80C18AB4 = { D_80C18A90, D_80C18AA8, 2 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/func_80C18928.s")
+static InitColliderStruct* D_80C18AC0[] = { &D_80C18A84, &D_80C18AB4 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/BgLastBwall_Update.s")
+static InitChainEntry D_80C18AC8[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Last_Bwall/BgLastBwall_Draw.s")
+void BgLastBwall_InitCollider(ColliderTrisInit* init, Vec3f* pos, Vec3s* rot, ColliderTris* collider,
+                              InitColliderStruct* arg4) {
+    s32 i;
+    s32 j;
+    Vec3f sp6C[3];
+    Vec3f sp60;
+    Vec3f sp54;
+
+    Matrix_RotateYS(rot->y, MTXMODE_NEW);
+    Matrix_RotateXS(rot->x, MTXMODE_APPLY);
+    Matrix_RotateZS(rot->z, MTXMODE_APPLY);
+    for (i = 0; i < init->count; i++) {
+        for (j = 0; j < 3; j++) {
+            Math_Vec3s_ToVec3f(&sp60, &arg4->unk0[arg4->unk4[j]]);
+            Matrix_MultVec3f(&sp60, &sp54);
+            Math_Vec3f_Sum(&sp54, pos, &sp6C[j]);
+        }
+
+        arg4->unk4 += 3;
+        Collider_SetTrisVertices(collider, i, &sp6C[0], &sp6C[1], &sp6C[2]);
+    }
+}
+
+void BgLastBwall_Init(Actor* thisx, PlayState* play) {
+    u32 pad;
+    BgLastBwall* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, D_80C18AC8);
+    this->unk238 = BGLASTBWALL_GET_F000(&this->dyna.actor);
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(play, &this->dyna, D_80C18A48[this->unk238].colHeader);
+    Collider_InitTris(play, &this->colliderTris);
+    if (Flags_GetSwitch(play, BGLASTBWALL_GET_SWITCHFLAGS(&this->dyna.actor))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else if (!Collider_SetTris(play, &this->colliderTris, &this->dyna.actor, &D_80C18A38,
+                                 &this->colliderTrisElement)) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        BgLastBwall_InitCollider(&D_80C18A38, &this->dyna.actor.world.pos, &this->dyna.actor.shape.rot,
+                                 &this->colliderTris, D_80C18AC0[this->unk238]);
+        SubS_FillCutscenesList(&this->dyna.actor, &this->nextCutscene, 1);
```

Comment:
```suggestion
        SubS_FillCutscenesList(&this->dyna.actor, this->cutscenes, ARRAY_COUNT(this->cutscenes));
```
From this function call, it looks nextCutscene should just be an array of length 1, and can probably just be called cutscenes.